BASE/src/ServerTelemetryChannel/Implementation/ErrorHandlingTransmissionPolicy.cs

```diff
@@ -122,6 +122,11 @@ private void HandleTransmissionSentEvent(object sender, TransmissionProcessedEve
                     TelemetryChannelEventSource.Log.TransmissionDataLossError(e.Transmission.Id,
                         "Unknown Exception Message");
                 }
+
+                if (e.Transmission.ManualFlushAsyncFlag)
```

Comment:
This check is unnecessary since it's done inside `SetFlushTaskCompletionSourceResult`

Reply:
Wanted to call SetFlushTaskCompletionSourceResult only if we have a special flag set in Transmission. As SetFlushTaskCompletionSourceResult is a public method and does want consumer to send a completion on Null object.

Reply:
I don't understand. Let me restate: The body of `SetFlushTaskCompletionSourceResult` already has the line `if (this.ManualFlushAsyncFlag)` and does nothing if the flag is false, then there's no point to testing the flag here. Just call `SetFlushTaskCompletionSourceResult` unconditionally.